### PR TITLE
util: Fix util_mr code check for FI_MR_SCALABLE

### PR DESCRIPTION
**Pull request by a-ilango**
- libfabric version 1.5 replaced FI_MR_SCALABLE with new mr_mode bits. Update the
util_mr code to handle it correctly.
- Update sockets provider to correctly set mr_mode based on api version

### CODE REVIEW

File: prov/util/src/util_mr.c [hunk 1]
```diff
@@ -152,14 +154,25 @@ static int compare_mr_keys(void *key1, void *key2)
...
 
-	map->mode = mode;
+	switch (mode) {
+	case FI_MR_BASIC:
+		map->mode = OFI_MR_BASIC_MAP;
+	case FI_MR_SCALABLE:
+		map->mode = 0;
+	default:
+		map->mode = mode;
+	}
```
Comment (shefty) [inline prov/util/src/util_mr.c]: There are no break statements in the switch cases.  On this note, the function is still taking the enum as input.  Is that correct, or should this be updated to accept flags?  If taking the enum is correct, I would change the map->mode assignment in the default case to "= 0", and assert that mode == UNSPEC.

Comment (a-ilango) [inline prov/util/src/util_mr.c]: >There are no break statements in the switch cases

Oops. Will fix that. 

I think the function needs to be updated to take flags. The default case would then be okay as it is.

**Comment by a-ilango** [comment]
This PR is ready for merge if there are no more comments.

File: prov/util/src/util_mr.c [hunk 2]
```diff
@@ -152,14 +154,27 @@ static int compare_mr_keys(void *key1, void *key2)
 }
 
 
-int ofi_mr_map_init(const struct fi_provider *prov, enum fi_mr_mode mode,
+/*
+ * If a provider or app whose version is < 1.5, calls this function and passes
+ * FI_MR_UNSPEC as mode, it would be treated as FI_MR_SCALABLE.
```
Comment (shefty) [inline prov/util/src/util_mr.c]: nit: please change FI_MR_SCALABLE to scalable.  The mode is mapped to 0, not the value of FI_MR_SCALABLE.

**Pull request merged**